interface/ceed-preconditioning.c

```diff
@@ -361,6 +362,30 @@ static inline int CeedSingleOperatorAssembleAddDiagonal_Core(CeedOperator op, Ce
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Core logic for assembling operator diagonal or point block diagonal
+
+  @param[in]  op             `CeedOperator` to assemble diagonal or point block diagonal
+  @param[in]  request        Address of @ref CeedRequest for non-blocking completion, else @ref CEED_REQUEST_IMMEDIATE
+  @param[in]  is_point_block Boolean flag to assemble diagonal or point block diagonal
+  @param[out] assembled      `CeedVector` to store assembled diagonal
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Developer
+**/
+static inline int CeedSingleOperatorLinearAssembleAddDiagonal(CeedOperator op, CeedRequest *request, const bool is_point_block,
+                                                              CeedVector assembled) {
+  Ceed ceed;
+  bool is_at_points;
+
+  CeedCall(CeedOperatorGetCeed(op, &ceed));
+  CeedCall(CeedOperatorIsAtPoints(op, &is_at_points));
+  CeedCheck(!is_at_points, ceed, CEED_ERROR_UNSUPPORTED, "Composite operator not supported");
```

Comment:
Should this read:
```suggestion
  CeedCheck(!is_at_points, ceed, CEED_ERROR_UNSUPPORTED, "AtPoints operator not supported");
```
?